### ISSUE TITLE
Fastlane snapshot UI test fails, but Xcode executes the UI test correctly.

### ISSUE THREAD

**Issue by GBoyadziev** [fastlane/fastlane#7809]
### New Issue Checklist

- [X] Updated fastlane to the latest version
- [X] I have read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)

### Issue Description
I am relatively new to fastlane, so sorry in advance :).
When I run my snapshot UI test ( for collecting screenshots ), when I run it via fastlane (with a defined lane ( see the environment for more info ) it fails, with error that it can't find a UI element. The logic of the test is to run several search queries using a search bar and make screenshots of the result. 

I am including the code from the UI test that generates the error. In it I am basically trying to get a search field, so I can start making searches with it. Because the record results have sometimes failed or produced different code snippets I have added several fallbacks linked to checks with `exists` method.
```
 let app = XCUIApplication()
// main way of getting search field
app.tabBars.buttons["Search"].tap()        
let navigationBar = app.navigationBars["listView"]
var searchField = navigationBar.searchFields["Search"]

// fallbacks
if !searchField.exists {
    searchField = navigationBar.searchFields["listSearchBar"]
    if !searchField.exists {
        searchField = navigationBar.otherElements["listSearchBar"]
        if !searchField.exists {
            let searchBarView = navigationBar.otherElements["listSerchBarView"]
            if !searchBarView.exists {
                searchField = searchBarView.searchFields["Search"]
                if !searchField.exists {
                    searchField = searchBarView.otherElements["listSearchBar"]
                }
            }
        }
    }
}
        
 XCTAssert(searchField.exists, "Could not find search field")
```
The fallbacks part of the code is not elegant at all :(, but it is only until I discover how to do it properly.      

The hierarchy is as follows I have a child view in the navigation bar, which is animated, when the user taps a search button, in it there is a search bar (listSearchBarView and listSearchBar are the accessibility identifiers for the parent view and the search bar ).

The error that i see in terminal is "UI Testing Failure - No matches found for "videoListSearchBarView" Other" and a failure in the assertion too, which is logical.

 Can someone explain whats the problem?

##### Complete output when running fastlane, including the stack trace and command used
> You can use: `--capture_output` as the last commandline argument to get that collected for you

<details><summary>🚫 fastlane environment 🚫</summary>

### Stack

| Key                 | Value                                                          |
| ------------------- | -------------------------------------------------------------- |
| OS                  | 10.12.2                                                        |
| Ruby                | 2.0.0                                                          |
| Bundler?            | false                                                          |
| Git                 | git version 2.10.1 (Apple Git-78)                              |
| Installation Source | /usr/local/bin/fastlane                                        |
| Host                | Mac OS X 10.12.2 (16C67)                                       |
| Ruby Lib Dir        | /System/Library/Frameworks/Ruby.framework/Versions/2.0/usr/lib |
| OpenSSL Version     | OpenSSL 0.9.8zc 19 Mar 2015                                    |
| Is contained        | false                                                          |
| Xcode Path          | /Applications/Xcode.app/Contents/Developer/                    |
| Xcode Version       | 8.2.1                                                          |

### System Locale

| Error                       |
| --------------------------- |
| No Locale with UTF8 found 🚫 |

### fastlane files:

<details><summary>`./fastlane/Fastfile`</summary>

```ruby
# Customise this file, documentation can be found here:
# https://github.com/fastlane/fastlane/tree/master/fastlane/docs
# All available actions: https://docs.fastlane.tools/actions
# can also be listed using the `fastlane actions` command

# Change the syntax highlighting to Ruby
# All lines starting with a # are ignored when running `fastlane`

# If you want to automatically update fastlane if a new version is available:
# update_fastlane

# This is the minimum version number required.
# Update this, if you use features of a newer version
fastlane_version "2.6.0"

default_platform :ios

platform :ios do
  before_all do
    # ENV["SLACK_URL"] = "https://hooks.slack.com/services/..."
    # cocoapods
  end

  desc "Runs all the tests"
  lane :test do
    scan
  end

  desc "Takes screenshots for specific IAP Products for both targets"
  lane :iapScreenshots do
    ["target1", "target2"].each do |scheme_name|
      ENV["SCHEME"] = scheme_name
      ENV["SCREENS_PATH"] = "./fastlane/screenshots/" + scheme_name
      clear_derived_data
      snapshot
    end
  end

  desc "Takes screenshots for specific IAP Products for target1 app"
  lane :iapScreenshotsTarget1 do
    ["target1"].each do |scheme_name|
      ENV["SCHEME"] = scheme_name
      ENV["SCREENS_PATH"] = "./fastlane/screenshots/" + scheme_name
      clear_derived_data
      snapshot
    end
  end

  desc "Takes screenshots for specific IAP Products for Target2"
  lane :iapScreenshotsTarget2 do
    ["target2"].each do |scheme_name|
      ENV["SCHEME"] = scheme_name
      ENV["SCREENS_PATH"] = "./fastlane/screenshots/" + scheme_name
      clear_derived_data
      snapshot
    end
  end

  # You can define as many lanes as you want

  after_all do |lane|
    # This block is called, only if the executed lane was successful

    # slack(
    #   message: "Successfully deployed new App Update."
    # )
  end

  error do |lane, exception|
    # slack(
    #   message: exception.message,
    #   success: false
    # )
  end
end


# More information about multiple platforms in fastlane: https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Platforms.md
# All available actions: https://docs.fastlane.tools/actions

# fastlane reports which actions are used
# No personal data is recorded. Learn more at https://github.com/fastlane/enhancer
```
</details>

**No Appfile found**

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.7.0   | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 1.5.1        |
| i18n                      | 0.7.0        |
| thread_safe               | 0.3.5        |
| tzinfo                    | 1.2.2        |
| minitest                  | 5.8.1        |
| activesupport             | 4.2.4        |
| CFPropertyList            | 2.3.4        |
| claide                    | 1.0.1        |
| colored                   | 1.2          |
| nanaimo                   | 0.2.3        |
| xcodeproj                 | 1.4.1        |
| rouge                     | 1.10.1       |
| xcpretty                  | 0.2.4        |
| terminal-notifier         | 1.6.3        |
| terminal-table            | 1.4.5        |
| plist                     | 3.1.0        |
| addressable               | 2.3.8        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| babosa                    | 1.0.2        |
| highline                  | 1.7.8        |
| commander                 | 4.4.3        |
| excon                     | 0.45.4       |
| faraday                   | 0.9.2        |
| unf_ext                   | 0.0.7.2      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20160128 |
| http-cookie               | 1.0.2        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 1.6.8        |
| gh_inspector              | 1.0.2        |
| uber                      | 0.0.15       |
| representable             | 2.3.0        |
| multi_json                | 1.11.2       |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2015.1120  |
| mime-types                | 3.0          |
| hurley                    | 0.2          |
| little-plugger            | 1.1.4        |
| logging                   | 2.0.0        |
| jwt                       | 1.5.2        |
| memoist                   | 0.14.0       |
| os                        | 0.9.6        |
| signet                    | 0.7.2        |
| googleauth                | 0.5.1        |
| thor                      | 0.19.1       |
| httpclient                | 2.7.1        |
| google-api-client         | 0.9.2        |
| mini_magick               | 4.5.1        |
| multi_xml                 | 0.5.5        |
| rubyzip                   | 1.1.7        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.1.1        |
| bundler                   | 1.13.7       |
| faraday_middleware        | 0.10.0       |
| json                      | 1.8.3        |
</details>

*generated on:* **2017-01-10**
</details>

**Comment by ohayon** [comment]
Hey @GBoyadziev, apologies for the delay getting to this issue! Are you still experiencing this problem? If so, could you show us the output from your command line when you run _snapshot_? 

Also, did you record this UI test using Xcode? Does the assertion fail when running the tests by just hitting command+U while in Xcode? 

Looking forward to hearing back! Thanks! 🐙 😄

**Comment by GBoyadziev** [comment]
Hello, today I updated to the latest version of fastlane and ran my snapshot lanes. Mostly everything works as expected, I got one fail, like the one mentioned above.
CommonUITests
✗ testInAPPScreenShots, XCTAssertTrue failed - Could not find search field
✗ testInAPPScreenShots, UI Testing Failure - No matches found for "videoListSearchBarView" Other

I think the issue has something to do with the accessibility framework, rather than fastlane. But the strange thing is that all of my UI tests pass when I run them from the Xcode.
